include/zephyr/device.h

```diff
@@ -197,7 +207,8 @@ typedef int16_t device_handle_t;
 			DEVICE_DT_NAME(node_id), init_fn, pm, data, config,    \
 			level, prio, api,                                      \
 			&Z_DEVICE_STATE_NAME(Z_DEVICE_DT_DEV_ID(node_id)),     \
-			__VA_ARGS__)
+			__VA_ARGS__);                                          \
+	Z_DEVICE_EXPORT(node_id)
```

Comment:
you can use `IF_ENABLED(CONFIG_LLEXT_EXPORT_DEVICES, (EXPORT_SYMBOL(DEVICE_DT_NAME_GET(node_id))))`

Reply:
I could, but the separate macro is easier to extend in the (short) future; I have a draft of a more complex proposal (https://github.com/zephyrproject-rtos/zephyr/pull/77799) where there are two different implementations for `Z_DEVICE_EXPORT`. 
For this reason I'm proposing to keep the separate macro, but included within an `IF_ENABLED` in the parent.